samples/StaticFilesAuth/Startup.cs

```diff
@@ -96,16 +109,27 @@ public void Configure(IApplicationBuilder app, IHostingEnvironment env, IAuthori
             
             var files = new PhysicalFileProvider(Path.Combine(env.ContentRootPath, "PrivateFiles"));
 
-            app.Map("/MapAuthenticatedFiles", branch =>
+            app.Use(async (context, next) =>
```

Comment:
This two stage process seems unnecessary. 
```
app.Map("/MapImperativeFiles", branch => {
  branch.Use((context, next) => { SetFileEndpoint(context, files, "files"); return next(); })
  branch.UseAuthorization();
  SetupFileServer(branch, files);
});
app.Map("/MapAuthenticatedFiles", branch => {
  branch.Use((context, next) => { SetFileEndpoint(context, files, null); return next(); })
  branch.UseAuthorization();
  SetupFileServer(branch, files);
});
```

Which could be consolidated to:
```
void MapFileAuth(IApplicationBuilder app, string path, IFileProvider files, string policy)
{
app.Map(path, branch => {
  branch.Use((context, next) => { SetFileEndpoint(context, files, policy); return next(); })
  branch.UseAuthorization();
  SetupFileServer(branch, files);
});
}

MapFileAuth(app, "/MapImperativeFiles", files, "files");
MapFileAuth(app, "/MapAuthenticatedFiles", files, null);
```

Reply:
Having a single UseAuthorization per branch makes sense, but don't artificially bend the pipeline back on itself just to reduce the calls to UseAuthorization.

Reply:
I do like where aspnet/Routing#903 is going, but we're not there yet.

Reply:
I agree the overlapping file structure is not realistic, but the branching is.